src/linter/root.go

```diff
@@ -1526,21 +1526,79 @@ func IsRuleEnabledForPath(root *RuleNode, filePath string, checkRule string) boo
 	// Starting with global state. We have guarantee while parsing config that rule is `on` and exist
 	ruleState := true
 
-	for _, part := range parts {
```

Comment:
in go `for-range` returns the index and the value of the elements in each index. Are you sure you should have changed it to `for-loop`?